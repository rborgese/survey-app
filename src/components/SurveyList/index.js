import * as React from 'react';
import { Link } from 'react-router-dom';

const SurveyList = ({ surveys }) => {
  return surveys ? (
    surveys.map((survey, index) => (
      <div key={index} className="col-lg-4 col-md-4 col-sm-4 col-xs-12">
        <div className="box-part text-center">
          <i
            className="glyphicon glyphicon-list-alt gi-3x"
            aria-hidden="true"
          />
          <div className="title">
            <h3>{survey.title} </h3>
          </div>
          <div className="text">{survey.tagline} </div>
          <Link to={`/surveys/${survey.id}`}>Take Survey</Link>
        </div>
      </div>
    ))
  ) : (
    <span className="text-center"> no surveys </span>
  );
};

export default SurveyList;
