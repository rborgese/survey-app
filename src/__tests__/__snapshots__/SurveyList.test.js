import React from 'react';
import renderer from 'react-test-renderer'
import SurveyList from '../../components/SurveyList'

// should render the ssurevylist
test('survey list renders correctly', () => {
    const component = renderer.create(<SurveyList/>)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
})

