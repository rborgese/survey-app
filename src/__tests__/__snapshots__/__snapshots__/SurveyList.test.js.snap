// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`survey list renders correctly 1`] = `
<span
  className="text-center"
>
   no surveys 
</span>
`;
